### ISSUE TITLE
Run phar tool directly

### ISSUE THREAD

**Issue by dmolineus** [phpcq/phpcq#18]
It should be possible to run a phar file to pass extra parameters, f.e.

`vendor/bin/phpcq exec phpunit tests/Example.php`

**Comment by dmolineus** [comment]
I'm thinking about it again. I'd like to avoid that we need to describe all arguments and options of the tool again. Also I don't want to use a hacky solution like https://stackoverflow.com/a/39400593.

What about generating a simple bash script for each installed in `vendor/phpcq` or `vendor/bin` directory which would call the installed tool directly?

```bash
#!/usr/bin/env php
<?php

function includeIfExists($file)
{
    return file_exists($file) ? include $file : false;
}
if ((!$loader = includeIfExists(__DIR__.'/vendor/autoload.php')) && (!$loader = includeIfExists(__DIR__.'/../../autoload.php'))) {
    echo 'You must set up the project dependencies, run the following commands:'.PHP_EOL.
        'curl -sS https://getcomposer.org/installer | php'.PHP_EOL.
        'php composer.phar install'.PHP_EOL;
    exit(1);
}

array_unshift($argv);
$command = array_merge(['/path/to/php', '/path/to/installed/tool.phar'], $argv);
// Exec command
```

Wdyt @discordier @baumannsven

**Comment by discordier** [comment]
This won't play along nicely if someone decides to still require any tool directly via composer.json (for whatever reason), as then there already exists a script with the given name.

**Comment by dmolineus** [comment]
That's why I also suggested to symlink ot to `vendor/phpcq`. Another option could be that the symlink is only created if not already exists.

Should it be possible to use a local installed tool instead of a phar based in phpcq? Phpcq could try to detect the tool in `vendor/bin`. If it exists there is no need to download a phar at all.

**Comment by discordier** [comment]
>That's why I also suggested to symlink ot to vendor/phpcq. Another option could be that the symlink is only created if not already exists.

Which is counter intuitive. I think I will give it a go as plain command, I see this as a minor issue anyway, so we could put this on low prio.

> Should it be possible to use a local installed tool instead of a phar based in phpcq? Phpcq could try to detect the tool in vendor/bin. If it exists there is no need to download a phar at all.

This will then open up a new can of worms, as the installed tool is not necessarily the same version as required in `phpcq.yaml`, yet it is likely that it is.
How shall we proceed there then? We would be in need of a completely new list of launcher scripts for this way of functionality.
I'd prefer to see these only as installed libraries and not tools but they have the side effect of installing a binary and therefore we must not interfere with these.

**Comment by discordier** [comment]
Implemented via #26
